R/teal_lockfile.R

```diff
@@ -23,92 +44,140 @@
 #'
 #' @seealso [renv::snapshot()], [renv::restore()].
 #'
-#' @return Nothing. This function is executed for its side effect of creating a lockfile used in the `teal` application.
+#' @return
+#'  `ExtendedTask` processing `renv` lockfile or `NULL` if lockfile has been provided through options
+#'  (skipping asynchronous process).
+#'
+#' @name teal_lockfile
+#' @rdname teal_lockfile
 #'
 #' @keywords internal
-teal_lockfile <- function() {
+NULL
+
+#' @rdname teal_lockfile
+#' @keywords internal
+teal_lockfile_invoke <- function() {
   lockfile_path <- "teal_app.lock"
+  shiny::onStop(function() file.remove(lockfile_path))
+
   # If user has setup the file, there is no need to compute a new one.
   user_lockfile <- getOption("teal.renv.lockfile", "")
   if (!identical(user_lockfile, "")) {
     if (file.exists(user_lockfile)) {
       file.copy(user_lockfile, lockfile_path)
-      return(invisible(NULL))
+      logger::log_trace('Lockfile set using option "teal.renv.lockfile" - skipping automatic creation.')
+      return(NULL)
     } else {
       stop("lockfile provided through options('teal.renv.lockfile') does not exist.")
     }
   }
 
-  if (!(is_in_test() || is_r_cmd_check())) {
-    old_plan <- future::plan()
-    # If there is already a parallel (non-sequential) backend, reuse it.
-    if (inherits(old_plan, "sequential")) {
-      future::plan(future::multisession, workers = 2)
-    }
+  process <- ExtendedTask$new(function(...) {
```

Comment:
```suggestion
  process <- shiny::ExtendedTask$new(function(...) {
```

Reply:
Is it necessary to prefix `ExtendedTask`? The whole `shiny` is imported and it's functions are not prefixed throuhgout.

Reply:
unsure, then I suppose not need for the prefix - it was just a live edit during a call to figure out what are the real imports/dependencies of `mirai` package